Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -166,6 +166,18 @@ public protocol PackageContainerProvider {
     ///
     /// - Throws: If the package container could not be resolved or loaded.
     func getContainer(for identifier: Container.Identifier) throws -> Container
+
+    /// This method is called when the resolver thinks it *may* need these containers soon i.e.
+    /// it can be used by provider to pre-compute data.
+    ///
+    /// For performance reasons, this method should return as soon as possible.
+    func mayNeed(containers: [Container.Identifier])
```

Comment:
I think it would be nice if this API offered a completion callback for then the repository is available (and the client can expect `getContainer` to be efficient.

Now that I think about it more, is there a good reason to phrase this as a `prefetch` versus just changing the resolve to make an asynchronous lookup request for the container, and then if it finds it *needs* the container blocking at that point?